geps/gep-713/index.md

```diff
@@ -3,1155 +3,1172 @@
 
-**Chihiro**: _Policy attachment is more complex than we’d like, yeah._ 😐
-_Look, how about roll back your `baker` change for now? We can get together in
-the morning and start sorting this out._
+Conflicts between metaresources ultimately affecting the same scope MUST be resolved into so-called [*Effective metaresources*](#effective-metaresources), according to some defined [*merge strategies*](#merge-strategies).
 
-Ana shakes her head and rolls back her edits to the `baker` Deployment, then
-sits looking out over the lake as the deployment progresses.
+#### Effective metaresources
 
-**Ana**: _Done. Are things happier now?_
+The DAG that represents the hierarchy of targetable objects works as a map to orderly resolve, for each [effective target](#declared-targets-versus-effective-targets), a combinatorial spec that is collectively defined by the set of metaresources affecting the target. This combinatorial spec of each effective target is referred to as the *Effective metaresource* (or *Effective policy*).
 
-**Chihiro**: _Looks like, thanks. Reckon you can get back to your sailboard._ 🙂
+The process of calculating Effective metaresources (Effective policies) consists of walking the hierarchy of target objects, from least specific to most specific (i.e., "top-down" or, equivalently, from the roots towards the leaves of the DAG of target objects) or from most specific to least specific ("bottom-up"), map reducing to a single metaresource spec each pair of metaresources adjacent to each other in the hierarchy, by applying at each step one of the supported [*merge strategies*](#merge-strategies) (described below), until no more than one spec remains for each effective target.
 
-Ana sighs.
+Example of Effective metaresources based on a hierarchy of Gateway API resources:
 
-**Ana**: _Wish I could. Wind’s died down, though, and it'll be dark soon.
-Just gonna head home._
+```mermaid
+---
+config:
+  look: handDrawn
+  theme: neutral
+---
+graph
+    gc@{ shape: rect, label: "GatewayClass 1" }
+    g@{ shape: rect, label: "Gateway 1" }
+    r1@{ shape: rect, label: "Route 1" }
+    r2@{ shape: rect, label: "Route 2" }
+    b@{ shape: rect, label: "Backend 1" }
+    m1@{ shape: stadium, label: "Metaresource 1" }
+    m2@{ shape: stadium, label: "Metaresource 2" }
 
-**Chihiro**: _Ouch. Sorry to hear that._ 😐
+    gc --> g
+    g --> r1
+    g --> r2
+    r1 --> b
+    r2 --> b
 
-One more look out at the lake.
+    m1 -.-> g
+    m2 -.-> r1
+```
 
-**Ana**: _Thanks for the help. Wish we’d found better answers._ 😢
+The above yields 2 Effective metaresources:
+- For `Route 1`: a combination of `Metaresource 1` and `Metaresource 2`
+- For `Route 2`: equal to `Metaresource 1`
 
-### The Problem, restated
-What this parable makes clear is that, in the absence of information about what
-Policy is affecting an object, it’s very easy to make poor decisions.
+#### Merge strategies
```

Comment:
I like the idea of merge strategies, but they're a massive addition to the scope of this GEP. I'd rather reserve this for a future addition to policy attachment. That would help keep the scope of what we graduate to GA in this PR very narrow and hopefully simplify the review of this PR. Ideally graduating to GA should be a process of refining existing concepts, not adding new ones.

Reply:
To be completely honest, without the concept of merge strategy I'm not quite sure what we'd be doing.

In my head, defaults, overrides _are_ merge strategies; just like their nuances of JSON patching versus full replacement of specs. 

Maybe that was me, but, in https://github.com/kubernetes-sigs/gateway-api/discussions/2927#discussioncomment-8991870, it felt like we wanted to consolidate Direct and Inherited classes around a well-defined concept of merge strategy.

Reply:
How about start with an ImplementationSpecific option?